source/user/mcscf.rst

```diff
@@ -1,57 +1,299 @@
+  For example:
+.. code-block:: python
+
+  ncas, nelecas = (6,8)
+  mycas = myhf.CASSCF(ncas, nelecas)
+
+
+2) Specifying the molecular orbital (MO) index of the active space orbitals you want. 
+  The user can "manually" selects the MO orbital indices (in a 1-based indexing scheme) and passes them to the ``sort_mo`` function.
+  See :source:`examples/mcscf/10-define_cas_space.py` for more details.
+
+.. code-block:: python
+
+  mycas = mcscf.CASSCF(myhf, 4, 4)
+  # Note sort_mo by default take the 1-based orbital indices.
+  mo = mycas.sort_mo([5,6,8,9])
+  mycas.kernel(mo)
+
+
+3) Specifying the number of orbitals in each symmetry group. 
+This strategy can occasionally be helpful when the initial guess orbitals are not easily identifiable.
+
+.. code-block:: python
+
+  mycas = mcscf.CASSCF(mf, 12, 12)
+  ncore = {'A1g':5, 'A1u':5}
+  ncas = {'A1g':2, 'A1u':2,'E1ux':1, 'E1uy':1, 'E1gx':1, 'E1gy':1,
+              'E2ux':1, 'E2uy':1, 'E2gx':1, 'E2gy':1}
+  mo = mcscf.sort_mo_by_irrep(mycas, mf.mo_coeff, ncas, ncore)
+  mycas.kernel(mo)
+
+A similar approach where we specify the electron occupations by irreducible representation is also possible by setting ``mycas.fcisolver.irrep_nelec``.
+
+.. code-block:: python
+
+  mycas = mcscf.CASSCF(myhf, 8, 8)
+  mycas.fcisolver.irrep_nelec = {"A1g": (2, 1), "A1u": (1, 1), "E1ux": (1, 1), "E1uy": (1, 0)}
+
+4) Use automated strategies (``avas`` and ``dmet_cas``) to pick an active space based on AO orbitals you're targeting.
+For more details, see :source:`examples/mcscf/43-avas.py` and :source:`examples/mcscf/43-dmet_cas.py`.
+
+.. code-block:: python
+
+  from pyscf.mcscf import avas
+  ao_labels = ['Fe 3d', 'Fe 4d', 'C 2pz']
+  ncas, nelecas, orbs = avas.avas(mf, ao_labels)
+  mycas = mcscf.CASSCF(mf, ncas, nelecas)
+
+  
+
+.. code-block:: python
+
+  from pyscf.mcscf import dmet_cas
+  ao_labels = ['Fe 3d', 'Fe 4d', 'C 2pz']
+  ncas, nelecas, mo = dmet_cas.guess_cas(mf, mf.make_rdm1(), ao_labels)
+  mycas = mcscf.CASSCF(mf, ncas, nelecas)
+  mycas.kernel(mo)
+
+
+Frozen Core MCSCF
+-----------------
+
+To reduce to computational expense of CASSCF calculations, users can "freeze" orbitals thereby excluding them from optimization.
+
+Users can specify a number to free the N lowest orbitals:
+
+.. code-block:: python
+
+  mycas = mcscf.CASSCF(myhf, 6, 8)
+  mycas.frozen = 2
+  mycas.kernel()
+
+
+Users can also specify a list of orbital indices (0-based).
+These may be occupied, virtual, or active orbitals.
+
+.. code-block:: python
+  mycas = mcscf.CASSCF(myhf, 6, 8)
+  mycas.frozen = [0,1,26,27]
+  mycas.kernel()
+
+See :source:`examples/mcscf/19-frozen_core.py` for a complete example.
+
+
+State-Averaged Calculations
+---------------------------
+
+When dealing with states that are close in energy, it can be helpful to perform state average calculations where the multireference orbitals are optimized for multiple states.
```

Comment:
I'm not sure whether it's suffciently clear that "states" mean "CAS wave functions".

There are no "multireference" orbitals; orbitals are always one-electron states.

Reply:
State-averaging is also needed to compute excited-states energies.  During SCF procedure all orbitals are optimized, not only active.

Reply:
@jamesETsmith yes, but "orbitals" don't mean they've been optimized in any way :)

Reply:
But yes, a better term would be CAS orbitals